File: navari_hotel_addon/public/js/taxes_and_totals.js
frappe.provide("erpnext.public");
frappe.provide("erpnext.controllers");

erpnext.taxes_and_totals.prototype.calculate_item_values = function() {
    var me = this;
    if (!this.discount_amount_applied) {
        for (const item of this.frm._items || []) {
            frappe.model.round_floats_in(item);
            item.net_rate = item.rate;
            item.qty = item.qty === undefined ? (me.frm.doc.is_return ? -1 : 1) : item.qty;

            if (!(me.frm.doc.is_return || me.frm.doc.is_debit_note)) {
                item.net_amount = item.amount = flt(item.rate * item.qty, precision("amount", item));
            }
            else {
                // allow for '0' qty on Credit/Debit notes
                let qty = flt(item.qty);
                if (!qty) {
                    qty = (me.frm.doc.is_debit_note ? 1 : -1);
                    if (me.frm.doc.doctype !== "Purchase Receipt" && me.frm.doc.is_return === 1) {
                        // In case of Purchase Receipt, qty can be 0 if all items are rejected
                        qty = flt(item.qty);
                    }
                }

                item.net_amount = item.amount = flt(item.rate * qty * item.days, precision("amount", item));
            }

            item.item_tax_amount = 0.0;
            item.total_weight = flt(item.weight_per_unit * item.stock_qty);

            me.set_in_company_currency(item, ["price_list_rate", "rate", "amount", "net_rate", "net_amount"]);
        }
    }
}

File: navari_hotel_addon/navari_hotel_addon/overrides/taxes_and_totals.py
from frappe.utils import flt
from erpnext.controllers.taxes_and_totals import calculate_taxes_and_totals


def calculate_item_values(self):
    if self.doc.get("is_consolidated") or self.discount_amount_applied:
        return

    do_not_round_fields = ["valuation_rate", "incoming_rate", "sales_incoming_rate"]
    for item in self._items:
        self.doc.round_floats_in(item, do_not_round_fields=do_not_round_fields)
        self.calculate_item_rate(item)

        item.net_rate = item.rate

        if (
            not item.qty
            and self.doc.get("is_return")
            and self.doc.get("doctype") != "Purchase Receipt"
        ):
            item.amount = flt(-1 * item.rate, item.precision("amount"))
        elif not item.qty and self.doc.get("is_debit_note"):
            item.amount = flt(item.rate, item.precision("amount"))
        else:
            item.amount = flt(
                item.rate * item.qty * item.days, item.precision("amount")
            )

        item.net_amount = item.amount

        self._set_in_company_currency(
            item,
            [
                "price_list_rate",
                "rate_with_margin",
                "rate",
                "net_rate",
                "amount",
                "net_amount",
            ],
        )

        item.item_tax_amount = 0.0


calculate_taxes_and_totals.calculate_item_values = calculate_item_values
